src/Components/DataHistoryExplorer/DataHistoryExplorer.tsx

```diff
@@ -0,0 +1,89 @@
+import React, { useState } from 'react';
+import {
+    IDataHistoryExplorerContext,
+    IDataHistoryExplorerProps,
+    IDataHistoryExplorerStyleProps,
+    IDataHistoryExplorerStyles
+} from './DataHistoryExplorer.types';
+import { getStyles } from './DataHistoryExplorer.styles';
+import {
+    classNamesFunction,
+    useTheme,
+    styled,
+    Stack,
+    FontIcon
+} from '@fluentui/react';
+import TimeSeriesBuilder from './Internal/TimeSeriesBuilder/TimeSeriesBuilder';
+import { IDataHistoryTimeSeriesTwin } from '../../Models/Constants';
+import { useTranslation } from 'react-i18next';
+
+export const DataHistoryExplorerContext = React.createContext<IDataHistoryExplorerContext>(
+    null
+);
+
+const getClassNames = classNamesFunction<
+    IDataHistoryExplorerStyleProps,
+    IDataHistoryExplorerStyles
+>();
+
+const DataHistoryExplorer: React.FC<IDataHistoryExplorerProps> = (props) => {
+    const { adapter, hasTitle = true, styles } = props;
+
+    // hooks
+    const { t } = useTranslation();
+
+    // state
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const [timeSeriesTwins, setTimeSeriesTwins] = useState<
+        Array<IDataHistoryTimeSeriesTwin>
+    >([]);
+
+    // styles
+    const classNames = getClassNames(styles, {
+        theme: useTheme()
+    });
+
+    return (
+        <DataHistoryExplorerContext.Provider
+            value={{
+                adapter
+            }}
+        >
+            <div className={classNames.root}>
+                {hasTitle && (
+                    <div className={classNames.titleWrapper}>
+                        <FontIcon
+                            className={classNames.titleIcon}
+                            iconName={'Chart'}
+                        />
+                        <span className={classNames.title}>
+                            {t('dataHistoryExplorer.title')}
+                        </span>
+                    </div>
+                )}
+                <Stack
+                    horizontal
+                    tokens={{ childrenGap: 8 }}
+                    className={classNames.contentStack}
+                >
+                    <TimeSeriesBuilder
+                        onTimeSeriesTwinListChange={(twins) =>
```

Comment:
can we avoid the full set operation here and instead have add/remove? that feels like the child has more granular control and doesn't just fully control the entire list.

Reply:
As I explained in another comment, there is no need to distinguish for add/remove here. onTimeSeriesTwinListChange is used to update the shared context with the viewer whenever the timeSeriesTwinList changes regardless add/remove. So, whenever the list is changed in the builder, the new list of series is passed to viewer through the shared context to render the series in chart.